scheduler/src/cook/mesos/api.clj

```diff
@@ -700,15 +753,38 @@
    :host-placement (make-default-host-placement)
    :straggler-handling default-straggler-handling})
 
+(defn validate-partitions
+  [{:keys [dataset partitions] :as in}]
+  (let [partition-type (get dataset "partition-type" nil)]
+    (cond
+      (nil? partition-type)
+      (when-not (empty? partitions)
+        (throw (ex-info "Dataset with partitions must supply partition-type"
+                        {:dataset dataset})))
+      (= "date" partition-type)
+      (doall (map (partial s/validate DatePartition) partitions))
```

Comment:
Why restrict date partitions to not contain keys other than `begin` and `end`?

Reply:
Discussed offline

Reply:
Done